dev-resources/template/2_lrs.yml

```diff
@@ -376,25 +382,36 @@ Resources:
         02_configureLrs:
           files:
             "/opt/lrsql/config/lrsql.json":
-              content: !Sub |
-                {
-                  "database": {
-                    "dbHost": "${DBHost}",
-                    "dbPort": ${DBPort},
-                    "dbName": "${DBName}",
-                    "dbUser": "${DBAppUserName}",
-                    "dbPassword": "${InitDBCustomResource.dbAppPass}"
-                  },
-                  "lrs" : {
-                    "adminUserDefault": "${DefaultAdminUser}",
-                    "adminPassDefault": "${DefaultAdminPass}",
-                    "authorityUrl": "http://mydomain.com"
-                  },
-                  "webserver": {
-                    "httpHost": "0.0.0.0",
-                    "httpPort": ${InstanceHttpPort}
+              content: !Sub
+                - |
+                  {
+                    "database": {
+                      "dbHost": "${DBHost}",
+                      "dbPort": ${DBPort},
+                      "dbName": "${DBName}",
+                      "dbUser": "${DBAppUserName}",
+                      "dbPassword": "${InitDBCustomResource.dbAppPass}"
+                    },
+                    "lrs" : {
+                      "adminUserDefault": "${DefaultAdminUser}",
+                      "adminPassDefault": "${DefaultAdminPass}",
+                      "authorityUrl": "http://mydomain.com"
+                    },
+                    "webserver": {
+                      "httpHost": "0.0.0.0",
+                      "httpPort": ${InstanceHttpPort},
+                      "allowedOrigins": ${AllowedOrigins}
+                    }
                   }
-                }
+                - AllowedOrigins: !If
+                    - SetCORS
+                    - !Sub
+                      - '["${JoinedAllowedOrigins}"]'
+                      - JoinedAllowedOrigins: !Join ['","', !Ref CORSAllowedOrigins]
+                    - !If
+                      - SetDNS
+                      - !Sub '["https://${ALBHostName}"]'
+                      - "[]"
```

Comment:
What exactly does this section mean in plain English?

Reply:
See the description above:

Reply:
> The logic is now as follows: If the template is given the CORSAllowedOrigins param, these are used. If not, the ALBHostname is used, if it is not present no web origins are allowed.

Reply:
This info is also provided in the parameter description.